Java/src/com/gymfox/echoserver/WazzupClient.java

```diff
@@ -0,0 +1,133 @@
+package com.gymfox.echoserver;
+
+import com.gymfox.communication.IPv4Address;
+
+import java.io.*;
+import java.net.Socket;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static com.sun.jndi.ldap.LdapCtx.DEFAULT_PORT;
+
+public class WazzupClient {
+    private final static int MAX_PERMISSION_PORT = 65536;
+    private final static int MIN_PERMISSION_PORT = 1024;
+    private Socket socket;
+    private IPv4Address address;
+    private int port;
+    private volatile boolean isConnect;
```

Comment:
Looks like it cannot be changed from the other thread. Thus no need in `volatile`